packages/nextjs/src/config/types.ts

```diff
@@ -467,6 +467,24 @@ export type SentryBuildOptions = {
    */
   suppressOnRouterTransitionStartWarning?: boolean;
 
+  /**
+   * Disables automatic injection of the route manifest into the client bundle.
+   *
+   * The route manifest is a build-time generated mapping of your Next.js App Router
+   * routes that enables Sentry to group transactions by parameterized route names
+   * (e.g., `/users/:id` instead of `/users/123`, `/users/456`, etc.).
+   *
+   * **Disable this option if:**
+   * - You want to minimize client bundle size
+   * - You're experiencing build issues related to route scanning
+   * - You're using custom routing that the scanner can't detect
+   * - You prefer raw URLs in transaction names
+   * - You're only using Pages Router (this feature is only supported in the App Router)
+   *
+   * @default false
+   */
+  disableManifestInjection?: boolean;
```

Comment:
l/feel free to ignore: I generally prefer `enableX` with default `true` over `disableX` with default `false` flag patterns but not sure if there's already precedence in other options here. To me the enableX pattern is a bit more intuitive from 
- a user perspective because they can switch on/off a feature without having to invert it in their mind
- a dev perspective because I don't have to invert the ifs (i.e. `if enableX doX()`).  

But again, this might be just me, so happy to leave this up to you and no objections to going with`disable` here.